### PR TITLE
Introduce android-base

### PR DESCRIPTION
**Pull request by mhamrle**
First part of https://github.com/monetas/gotary/issues/422

### CODE REVIEW

**Comment by haarts** [comment]
This PR is like the sun after a late summer morning shower. :dancers:

File: android-base/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+RUN wget http://dl.google.com/android/$ANDROID_IMAGE && \
+    tar -C /opt/ -xzf $ANDROID_IMAGE && \
+    rm $ANDROID_IMAGE
+
+# Add android tools and platform tools to PATH
+ENV ANDROID_HOME /opt/android-sdk-linux
+ENV PATH $PATH:$ANDROID_HOME/tools
+ENV PATH $PATH:$ANDROID_HOME/platform-tools
+
+#Install gradle
```
Comment (toxeus) [inline android-base/Dockerfile]: Shouldn't this be part of java-base?

File: android-base/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+FROM monetas/java-base:4
```
Comment (toxeus) [inline android-base/Dockerfile]: `FROM monetas/java-base:5`

File: android-base/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ENV GRADLE_DIR gradle-2.4
+ENV GRADLE_IMAGE $GRADLE_DIR-all.zip
+RUN wget https://services.gradle.org/distributions/$GRADLE_IMAGE && \
+    unzip -d /opt/ $GRADLE_IMAGE && \
+    rm $GRADLE_IMAGE
+ENV GRADLE_HOME /opt/$GRADLE_DIR
+ENV PATH $GRADLE_HOME/bin:$PATH
+
+# Install android sdk parts
+RUN (while true; do echo "y"; sleep 2; done) | android update sdk -a --filter android-22,build-tools-22.0.1,extra-android-m2repository --no-ui
```
Comment (toxeus) [inline android-base/Dockerfile]: Any reason why you're not using `yes`?

Comment (mhamrle) [inline android-base/Dockerfile]: The `sleep 2` is needed.

Comment (toxeus) [inline android-base/Dockerfile]: Then, maybe a comment is appropriate ;)

File: android-base/Dockerfile (hunk 1 above)
Comment (mhamrle) [inline android-base/Dockerfile]: ok, I move it into java-base, hope it is not problem that there will be newer gradle

Comment (toxeus) [inline android-base/Dockerfile]: I don't know. I just thought that it makes sense to use the same gradle everywhere.

Comment (OttoAllmendinger) [inline android-base/Dockerfile]: @mhamrle 

> hope it is not problem that there will be newer gradle

I'm using gradle-2.4 on my system

File: android-base/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
+FROM monetas/java-base:6
```
Comment (toxeus) [inline android-base/Dockerfile]: This image does not yet exist. I expect the CI to fail.

**Comment by toxeus** [comment]
`wget` not found

File: android-base/Dockerfile (hunk 4 above)
Comment (toxeus) [inline android-base/Dockerfile]: or maybe not because this tag is also used locally

**Comment by mhamrle** [comment]
I am changing java-base and android-base in the same PR. Problem is that first is built android-base.

**Comment by toxeus** [comment]
@mhamrle you need to `apt-get install wget` in the java-base image.

**Comment by mhamrle** [comment]
`wget` is installed in java-base, but now when the images is build in reverse order android-base is using old java-base without `wget`

**Comment by toxeus** [comment]
Now I got it :smile:

File: android-base/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    tar -C /opt/ -xzf $ANDROID_IMAGE && \
+    rm $ANDROID_IMAGE
+
+# Add android tools and platform tools to PATH
+ENV ANDROID_HOME /opt/android-sdk-linux
+ENV PATH $PATH:$ANDROID_HOME/tools
+ENV PATH $PATH:$ANDROID_HOME/platform-tools
+
+# Install android sdk parts
+# Andoid needs to confirm licences. It is not possible to use `yes` command because sleep is required.
```
Comment (toxeus) [inline android-base/Dockerfile]: s/Andoid/Android/

**Comment by toxeus** [comment]
:+1: apart from the typo

**Pull request merged**